tools/unix/version.sh

```diff
@@ -9,8 +9,10 @@ set -euo pipefail
 # number of today's commits stays the same.
 # Force git with TZ variable and local dates to print the UTC date.
 COUNT_AND_DATE=( $(TZ=UTC0 git log --pretty=format:%cd --date=iso-local --since="30 days ago" | cut -d' ' -f 1 | sed 's/-/./g' | sort | uniq -c | tail -1) )
-if [ -z "$COUNT_AND_DATE" ]; then
+if [ ${#COUNT_AND_DATE[@]} -eq 0 ]; then
```

Comment:
Why is this syntax better than
```suggestion
if [ -z "${COUNT_AND_DATE:-}" ]; then
```
?

Reply:
Small detail, but we don't want `COUNT_AND_DATE` to be a scalar string, but an array of strings!
If the next contributor mistakenly changes line 11, and turns `COUNT_AND_DATE` into  a scalar string,
then `[ ${#COUNT_AND_DATE[@]} -eq 0 ];` catches that, but `-z "${COUNT_AND_DATE:-}"` does not, and the code only fails later.

On the other hand `-z "${COUNT_AND_DATE:-}"` is shorter, which is a pro.

Reply:
Good catch, thanks for the explanation! WDYT, can we improve the above command to get the appropriate version (the date of the last commit, and the number of commits on that day) without that "no commits in the last 30 days" issue?

Reply:
Should be possible. I will look into it.

Reply:
@Ferenc- also, as an additional improvement, it would be great to detect the git repo presence and also not fail if someone builds the code from an archive.

Reply:
Done